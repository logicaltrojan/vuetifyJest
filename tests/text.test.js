
import { mount } from '@vue/test-utils'
import myText from '@/components/Mytext.vue'


describe('myText', ()=> {
    test('is a vue component', ()=>{
        const wrapper = mount(myText);
        expect(wrapper.isVueInstance()).toBeTruthy();
    })

    test('contains text field ', ()=> {
        const wp = mount(myText);
        expect(wp.html()).toContain('<input')
    })

    test('contains button', ()=> {
        const wp = mount(myText);
        expect(wp.html()).toContain('<button')
    })

    test('renders prop correctly', ()=>{
        const testData = "hello2132312"
        const wp = mount(myText, { 
            propsData : {
                textData : testData
            }
        });
        
        expect(wp.vm.childData).toBe(testData);
        expect(wp.html()).toMatchSnapshot();
    })

    test('input makes diffrent', () => {
        var parent = mount({
            template : '<div><my-text :textData="parentData" @confirm="meme"></my-text></div>',
            data(){ 
                return {
                 parentData : "testData23"
                }
            },
            methods : {
                meme(val){
                    this.parentData = val;
                }
            },
            components : { 'my-text' : myText}
        })

        expect(parent.find('input')).toBeTruthy();
        var innerInputComponent = parent.find('input')
        const changTestData = 'changedInput'
        innerInputComponent.element.value = changTestData;
        innerInputComponent.trigger('input');

        parent.find('button').trigger('click');
        expect(parent.vm.parentData).toBe(changTestData);

        
    })

})