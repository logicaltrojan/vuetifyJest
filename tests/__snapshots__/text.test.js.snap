// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`myText renders prop correctly 1`] = `
<div>
  <div class="v-input v-input--is-label-active v-input--is-dirty theme--light v-text-field">
    <div class="v-input__control">
      <div class="v-input__slot">
        <div class="v-text-field__slot"><input id="input-17" type="text"></div>
      </div>
      <div class="v-text-field__details">
        <div class="v-messages theme--light"><span name="message-transition" tag="div" class="v-messages__wrapper"></span></div>
      </div>
    </div>
  </div> <button>ToCONFIRM</button>
</div>
`;
